File: 신선영/Week_14/P43238_입국심사.java
import java.util.*;

class Solution {
    public long solution(int n, int[] times) {
        Arrays.sort(times);
        long answer = 0;
        long left = 0;
        long right = times[times.length - 1] * (long) n;
        
        while (left <= right) {
            long mid = (left + right) / 2;
            long cnt = 0;   // 심사 받을 수 있는 사람 수
        
            for (int t : times) {
                cnt += mid / t;
            }
            
            if (cnt >= n) {
                answer = mid;
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        
        }
       return answer;
    }
}

File: 이우엽/Week_14/P43238_입국심사.py
def solution(n, times):
    answer = 0
    left = 1
    right = max(times) * n

    while left < right:
        mid = (left + right) // 2
        total = 0

        for time in times:
            total += mid // time

        if total >= n:
            right = mid
        else:
            left = mid + 1
    answer = left
    return answer